common/flatpak-dir.c

```diff
@@ -14075,20 +14096,17 @@ flatpak_dir_get_default_locale_languages (FlatpakDir *self)
   g_ptr_array_sort (langs, flatpak_strcmp0_ptr);
   g_ptr_array_add (langs, NULL);
 
-  return (char **) g_ptr_array_free (g_steal_pointer (&langs), FALSE);
+  return flatpak_strv_merge(default_languages, (char **) g_ptr_array_free (g_steal_pointer (&langs), FALSE));
 }
 
 char **
 flatpak_dir_get_locale_languages (FlatpakDir *self)
 {
-  GKeyFile *config = flatpak_dir_get_repo_config (self);
+  char **langs = NULL;
 
-  if (config)
-    {
-      char **langs = g_key_file_get_string_list (config, "core", "xa.languages", NULL, NULL);
-      if (langs)
-        return sort_strv (langs);
-    }
+  langs = flatpak_dir_get_language_config_key (self, "xa.languages");
+  if (langs)
```

Comment:
Maybe add a comment that an empty array is intended to mean "all languages", and direct people to the `flatpak-config` man page?